src/lib/define-dynamic-block.js

```diff
@@ -168,6 +203,14 @@ const defineDynamicBlock = (ScratchBlocks, categoryInfo, staticBlockInfo, extend
                 const arg = blockInfo.arguments[argName];
                 switch (arg.type) {
                 case ArgumentType.STRING:
+                    if (arg.menu && !this.menus[arg.menu].acceptReporters) {
+                        const options = this.menus[arg.menu].items;
+                        args.push({type: 'field_dropdown', name: argName, options: options});
```

Comment:
Add support for non-droppable menus in dynamic extension blocks. All variable blocks/list blocks are now dynamic, so that we can dynamically update the block instance when a variable is renamed.

Reply:
🎉

To celebrate I'ma make a block where the menu flips between droppable and non-droppable based on which item is selected in the menu! ;)